WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalScanningForReader.swift

```diff
@@ -81,30 +81,45 @@ private extension CardPresentModalScanningForReader {
 
     enum Localization {
         static let title = NSLocalizedString(
-            "Scanning for reader",
+            "cardPresent.modalScanningForReader.title",
+            value: "Scanning for reader",
             comment: "Title label for modal dialog that appears when searching for a card reader"
         )
 
         static let instruction = NSLocalizedString(
-            "To turn on your card reader, briefly press its power button.",
+            "cardPresent.modalScanningForReader.instruction",
+            value: "To turn on your card reader, briefly press its power button.",
             comment: "Label within the modal dialog that appears when searching for a card reader"
         )
 
         static let cancel = NSLocalizedString(
-            "Cancel",
+            "cardPresent.modalScanningForReader.cancelButton",
+            value: "Cancel",
             comment: "Label for a cancel button"
         )
 
         static let learnMoreLink = NSLocalizedString(
-            "Learn more",
+            "cardPresent.modalScanningForReader.learnMore.link",
+            value: "Learn more",
             comment: """
                      A label prompting users to learn more about In-Person Payments.
                      This is the link to the website, and forms part of a longer sentence which it should be considered a part of.
                      """
         )
 
+        // \u{2011} (Non-Breaking Hyphen) will result in `genstrings` logging
+        //
+        // invalid unicode sequence: \u{201
+        //
+        // This is a known bug in the toolchain.
+        // The only known workaround is to use the actual character, but that would defeat the non-breaking purpose.
```

Comment:
For the record, I double-checked if using the `-u` flag of `genstrings` would help fix that issue (despite the man page saying that this flag is more to "Allow unicode characters in the value of strings files." rather than about parsing Unicode characters in source code) and confirmed it didn't make the issue go away.

> The only known workaround is to use the actual character, but that would defeat the non-breaking purpose.

I'm not sure I'm following what would be so wrong about using the actual "Non-Breaking Hyphen" literal character in the source code directly 🤔 
Is it to make sure that the translators also use a non-breaking hyphen in their translations (as in: there's more chances that they keep the `\u{2011}` verbatim in their translation, compared to if we used a `‑` literal in the source copy and let them re-type the `In‑Person` text as part of their translation)?

But I'm guessing even the `In‑Person` text will be translated in something else in the other locales anyway, so each locale should use the appropriate word for it and that word might not contain a hyphen in other locales to begin with…

If that's so, I'm not sure I agree with the rationale _not_ to use `‑` literal in the first place, and I think we might be better off using `‑` literal in the `value:`, then replace the `\u{2011} is a special character that acts as nonbreaking hyphen for "-" in the "In-Person" string.` mention in the `comment:` with something like `The hyphen in the "In‑Person" string is a non-breaking hyphen (U+2011). If your translation of that term also happens to contains a hyphen, please be sure to use a non-breaking hyphen character for it too.`

Reply:
@iamgabrielma To be clear, I'm not advocating we replace the non-breaking-hyphen with a regular hyphen. I'm suggesting we type the non-breaking-hyphen literal directly in the code, verbatim, instead of representing it using `\u{2011}`.

The end result would be completely unchanged, since the `String` would still contain a non-breaking-hyphen literal just like before.
It's just that by avoiding to use the `\u{2011}` Swift escape code for that character, and instead using the literal for non-breaking-hyphen directly, we will avoid encountering the bug with `genstrings` that doesn't seem to like `\u{` when parsing the code for `NSLocalizedString`.

Reply:
Gotcha! I miss-understood on my first reading. That is even better, thanks! 🙇